File: Java/StrategyPattern/src/Duck/CityDuck.java
package Duck;

import DuckBehaviours.Fly.IFlyBehaviour;
import DuckBehaviours.Quack.IQuackBehaviour;

public class CityDuck extends Duck{

    public CityDuck(IFlyBehaviour flyBehaviour, IQuackBehaviour quackBehaviour) {
        super(flyBehaviour, quackBehaviour);
    }

    public void performFly(){
        flyBehaviour.fly();
    }

    public void performQuack(){
        quackBehaviour.quack();
    }
}

File: Kotlin/StrategyPattern/src/ducks/CityDuck.kt
package ducks

import duckBehaviours.flyingBehaviour.IFlyingBehaviour
import duckBehaviours.quackBehaviour.IQuackBehaviour

class CityDuck(val flyingBehaviour: IFlyingBehaviour, val quackBehaviour: IQuackBehaviour) : Duck(flyingBehaviour, quackBehaviour){
    override fun performFly() {
        flyingBehaviour.fly()
    }

    override fun performQuack() {
        quackBehaviour.quack()
    }
}